THIRD-PARTY-NOTICES.txt

```diff
@@ -0,0 +1,213 @@
+This project uses third-party libraries or other resources that may be
+distributed under licenses different than the rest of the project.
+
+In the event that we accidentally failed to list a required notice, please
+bring it to our attention by submitting a GitHub issue.
+
+The attached notices are provided for information only.
+
+License notice for source files in the externals directory which were all taked from
+https://github.com/aspnet/AspNetCore/tree/e0e9096af526448cb1a02f97e60efd8567b7ba35
```

Comment:
Heads up @tmds. I was asked to add a a third party notices file if there wasn't one in the repo already. I mostly tried copying the format from the AspNetCore repo. No one has reviewed this file yet, so please feel free to suggest whatever changes you want here.

Reply:
If it's the same for you, I'd like to do something similar as what is in corefx:

THIRD-PARTY-NOTICES.TXT file:

https://github.com/dotnet/corefx/blob/ba50ba15d0747d5be1e8bd38ea03a5ba892314bd/THIRD-PARTY-NOTICES.TXT#L1-L18

header of file copied from ASP.NET Core:
https://github.com/dotnet/corefx/blob/ba50ba15d0747d5be1e8bd38ea03a5ba892314bd/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/DynamicTable.cs#L1-L3

Do we need this externals directory?

Reply:
.NET Core by kestrel-linux-transport, and dotnet@microsoft.com by dotnet-team@redhat.com.